compass_app/app/lib/common/utils/result.dart

```diff
@@ -0,0 +1,38 @@
+/// Utility class to wrap result data
+sealed class Result<T> {
+  const Result();
+
+  /// Creates an instance of Result containing a value
+  factory Result.ok(T value) => Ok(value);
+
+  /// Create an instance of Result containing an error
+  factory Result.error(Exception error) => Error(error);
+
+  /// Convenience method to cast to Ok
+  Ok<T> get asOk => this as Ok<T>;
+
+  /// Convenience method to cast to Error
+  Error get asError => this as Error<T>;
```

Comment:
Should there be `isOk` and `isError` boolean getters, as well? Either way, usage docstring to suggest the right way to interrogate a yet-ambiguous `Result` instance would be nice.

Reply:
I see later on that, because this is a sealed class, pattern matching works for this. That's great! A quick docstring suggesting that would still be useful, IMO :)

Reply:
Good point! I will add that.